packages/storage/src/indexed-db/index.ts

```diff
@@ -264,4 +271,63 @@ export class IndexedDb implements IndexedDbInterface {
   async saveGasPrices(value: GasPrices): Promise<void> {
     await this.u.update({ table: 'GAS_PRICES', value, key: 'gas_prices' });
   }
+
+  async getNotesForVoting(
+    addressIndex: AddressIndex | undefined,
+    votable_at_height: bigint,
+  ): Promise<NotesForVotingResponse[]> {
+    const relevantAssets = new Map<string, DenomMetadata>();
+
+    let assetCursor = await this.db.transaction('ASSETS').store.openCursor();
+
+    while (assetCursor) {
+      const isDelegationAssetType = new RegExp('udelegation_.*');
+
+      const denomMetadata = DenomMetadata.fromJson(assetCursor.value);
+
+      if (isDelegationAssetType.test(denomMetadata.base) && denomMetadata.penumbraAssetId) {
+        relevantAssets.set(uint8ArrayToHex(denomMetadata.penumbraAssetId.inner), denomMetadata);
+      }
+      assetCursor = await assetCursor.continue();
+    }
+
+    const notesForVoting = [];
+
+    let noteCursor = await this.db.transaction('SPENDABLE_NOTES').store.openCursor();
+    while (noteCursor) {
+      const note = SpendableNoteRecord.fromJson(noteCursor.value);
+
+      if (addressIndex) {
+        if (!note.addressIndex?.equals(addressIndex)) {
+          noteCursor = await noteCursor.continue();
+          continue;
+        }
+      }
+
+      if (!note.note?.value?.assetId?.inner) {
+        noteCursor = await noteCursor.continue();
+        continue;
+      }
+
+      const isRelevantAsset = relevantAssets.has(uint8ArrayToHex(note.note.value.assetId.inner));
+      const noteIsVotable = note.heightSpent === 0n || note.heightSpent > votable_at_height;
+
+      if (isRelevantAsset && noteIsVotable && note.heightCreated < votable_at_height) {
+        const asset = relevantAssets.get(uint8ArrayToHex(note.note.value.assetId.inner));
+
+        const bech32idk = asset?.base.replace('udelegation_', '');
+        if (bech32idk) {
```

Comment:
Probably should throw here instead:

```typescript
if (!bech32idk) throw new Error("expected delegation token identity key not present")
```

Reply:
In this case one broken note will break the query, but I think this is correct, we shouldn't return data if some of it is broken